File: samples/grpc/restaurant-drone-deliveries-service-scala/src/main/scala/central/DeliveriesSettings.scala
package central

import akka.actor.typed.ActorSystem

import scala.concurrent.duration.FiniteDuration
import scala.jdk.CollectionConverters._
import scala.jdk.DurationConverters._

object DeliveriesSettings {
  def apply(system: ActorSystem[_]): DeliveriesSettings = {
    val config =
      system.settings.config.getConfig("restaurant-drone-deliveries-service")
    val locationIds =
      config.getStringList("local-drone-control.locations").asScala.toSet
    val droneAskTimeout = config.getDuration("drone-ask-timeout").toScala
    val restaurantDeliveriesAskTimeout =
      config.getDuration("restaurant-deliveries-ask-timeout").toScala

    DeliveriesSettings(
      locationIds,
      droneAskTimeout,
      restaurantDeliveriesAskTimeout)
  }
}

final case class DeliveriesSettings(
    locationIds: Set[String],
    droneAskTimeout: FiniteDuration,
    restaurantDeliveriesAskTimeout: FiniteDuration)

File: samples/grpc/restaurant-drone-deliveries-service-java/src/main/java/central/DeliveriesSettings.java
package central;

import akka.actor.typed.ActorSystem;

import java.time.Duration;
import java.util.HashSet;
import java.util.Set;

public final class DeliveriesSettings {
  public final Set<String> locationIds;
  public final Duration droneAskTimeout;
  public final Duration restaurantDeliveriesAskTimeout;


  public DeliveriesSettings(Set<String> locationIds, Duration droneAskTimeout, Duration restaurantDeliveriesAskTimeout) {
    this.locationIds = locationIds;
    this.droneAskTimeout = droneAskTimeout;
    this.restaurantDeliveriesAskTimeout = restaurantDeliveriesAskTimeout;
  }

  public static DeliveriesSettings create(ActorSystem<?> system) {
    var config = system.settings().config().getConfig("restaurant-drone-deliveries-service");
    var locationIds = new HashSet<>(config.getStringList("local-drone-control.locations"));
    var droneAskTimeout = config.getDuration("drone-ask-timeout");
    var restaurantDeliveriesAskTimeout = config.getDuration("restaurant-deliveries-ask-timeout");
    return new DeliveriesSettings(locationIds, droneAskTimeout, restaurantDeliveriesAskTimeout);
  }
}
